### DISCUSSION TITLE
Microsoft Authentication provider not available in Codespaces

### DISCUSSION THREAD

**Discussion by AndreyTretyak** [community/community#46446] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I'm redirected from this issue https://github.com/microsoft/vscode/issues/173467

When using GitHub Codespaces I'm unable to sign in with Microsoft account:
<img width="254" alt="image" src="https://user-images.githubusercontent.com/2550113/216806867-1beceb8e-7558-4e72-85c1-bac354d6b318.png">
My extension that uses Microsoft authentication also unable to work in codespaces because of that. Adding `microsoft-authentication` to the `extensionDependencies` does not work also, I'm getting the error that `microsoft-authentication` not found. How can I get it in the Codespaces?

Version: 1.75.0
Commit: e2816fe719a4026ffa1ee0189dc89bdfdbafb164
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36 Edg/109.0.1518.70
Embedder: codespaces

**Comment by joshaber** [comment, marked as answer]
Hi @AndreyTretyak, the extension itself should be included by default, though it's not signed in by default:

<img width="1312" alt="image" src="https://user-images.githubusercontent.com/13760/217543915-2021b7e4-97be-41fe-8607-c60afae326c4.png">

To sign in I went to settings sync:

<img width="311" alt="image" src="https://user-images.githubusercontent.com/13760/217544197-7fcb50c6-fd12-453e-9189-e60c12fe3df1.png">

And chose the Microsoft identity:

<img width="626" alt="image" src="https://user-images.githubusercontent.com/13760/217544306-0bc7cd67-fe54-43d4-9145-23f3adedf5da.png">

After that the codespace refreshed and I can see that I'm signed in:

<img width="429" alt="image" src="https://user-images.githubusercontent.com/13760/217544538-66e16df6-929b-4bd8-a940-1607f6105223.png">

**Reply by AndreyTretyak** [reply]
Thank you, Josh, I'm sorry, it looks like it was a mistake on my part. I appreciate detailed explanations.
